modules/core/core-repo/src/main/java/com/enonic/xp/repo/impl/elasticsearch/query/translator/factory/dsl/SimpleStringQueryBuilder.java

```diff
@@ -0,0 +1,57 @@
+package com.enonic.xp.repo.impl.elasticsearch.query.translator.factory.dsl;
+
+import java.util.List;
+
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.SimpleQueryStringBuilder;
+
+import com.enonic.xp.data.PropertySet;
+import com.enonic.xp.repo.impl.elasticsearch.query.translator.factory.function.WeightedQueryFieldNames;
+import com.enonic.xp.repo.impl.elasticsearch.query.translator.resolver.SearchQueryFieldNameResolver;
+
+import static com.google.common.base.Strings.nullToEmpty;
+
+abstract class SimpleStringQueryBuilder
+    extends DslQueryBuilder
+{
+    protected final String query;
+
+    protected final String operator;
+
+    protected final WeightedQueryFieldNames fields;
+
+    protected final SearchQueryFieldNameResolver nameResolver;
+
+    SimpleStringQueryBuilder( final PropertySet expression )
+    {
+        super( expression );
+
+        this.query = getString( "query" );
+        this.operator = getString( "operator" );
+        final String field = getString( "field" );
```

Comment:
It is confusing to support both singular and plural form `fields`. What is the idea here?